From f85e2b2390a42bb43d91374ff57df83bca1491ae Mon Sep 17 00:00:00 2001
From: Zeng Heng <zengheng4@huawei.com>
Date: Sat, 25 Oct 2025 14:52:57 +0800
Subject: [PATCH 782/960] downstream: arm64/mpam: Print MPAM register operation
 for debug

To facilitate scoping the problem and making diagnostics easier,
MPAM provides print logs for register operations.

Enable this feature with the following command:

  echo 'file mpam_devices.c +p' > /sys/kernel/debug/dynamic_debug/control
  echo 8 > /proc/sys/kernel/printk

Signed-off-by: Zeng Heng <zengheng4@huawei.com>
Signed-off-by: Quanmin Yan <yanquanmin1@huawei.com>
Signed-off-by: Liu Shixin <liushixin2@huawei.com>
---
 drivers/platform/mpam/mpam_devices.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/platform/mpam/mpam_devices.c b/drivers/platform/mpam/mpam_devices.c
index 3656f3be6e91..3ade552d1d28 100644
--- a/drivers/platform/mpam/mpam_devices.c
+++ b/drivers/platform/mpam/mpam_devices.c
@@ -114,12 +114,16 @@ static void __mpam_write_reg(struct mpam_msc *msc, u16 reg, u32 val)
 	lockdep_assert_held_once(&msc->part_sel_lock);	\
 	____ret = __mpam_read_reg(msc, MPAMF_##reg);	\
 							\
+	pr_debug("par r: msc %d reg 0x%x val 0x%x\n",	\
+		  msc->id, MPAMF_##reg, ____ret);	\
 	____ret;					\
 })
 
 #define mpam_write_partsel_reg(msc, reg, val)			\
 ({								\
 	lockdep_assert_held_once(&msc->part_sel_lock);		\
+	pr_debug("par w: msc %d reg 0x%x val 0x%lx\n",		\
+		  msc->id, MPAMCFG_##reg, (unsigned long)val);	\
 	__mpam_write_reg(msc, MPAMCFG_##reg, val);		\
 })
 
@@ -130,12 +134,16 @@ static void __mpam_write_reg(struct mpam_msc *msc, u16 reg, u32 val)
 	lockdep_assert_held_once(&msc->mon_sel_lock);	\
 	____ret = __mpam_read_reg(msc, MSMON_##reg);	\
 							\
+	pr_debug("mon r: msc %d reg 0x%x val 0x%x\n",	\
+		  msc->id, MSMON_##reg, ____ret);	\
 	____ret;					\
 })
 
 #define mpam_write_monsel_reg(msc, reg, val)			\
 ({								\
 	lockdep_assert_held_once(&msc->mon_sel_lock);		\
+	pr_debug("mon w: msc %d reg 0x%x val 0x%lx\n",		\
+		  msc->id, MSMON_##reg, (unsigned long)val);	\
 	__mpam_write_reg(msc, MSMON_##reg, val);		\
 })
 
@@ -471,6 +479,9 @@ static int mpam_ris_get_affinity(struct mpam_msc *msc, cpumask_t *affinity,
 
 	cpumask_and(affinity, affinity, &msc->accessibility);
 
+	pr_debug("comp_id %d msc->id %d affinity %*pb\n",
+		  comp->comp_id, msc->id, cpumask_pr_args(affinity));
+
 	return 0;
 }
 
-- 
2.43.0

